### PR TITLE
filter: import the new table filter

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
import the new table filter

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- import the new filter

### CODE REVIEW

File: pkg/config/filter.go [hunk 1]
```diff
@@ -15,12 +15,11 @@ package config
 
 import (
 	"github.com/pingcap/parser/model"
-	"github.com/pingcap/tidb-tools/pkg/filter"
 )
 
 // FilterConfig represents filter config for a changefeed
 type FilterConfig struct {
-	*filter.Rules
+	Rules            []string           `toml:"rules" json:"rules"`
```
Comment (overvenus) [inline pkg/config/filter.go:22]: Please make it compatible (don't panic) with older releases.

File: cdc/owner.go [hunk 2]
```diff
@@ -294,6 +294,47 @@ func (o *Owner) newChangeFeed(
 	return cf, nil
 }
 
+func (o *Owner) checkAndCleanTasksInfo(ctx context.Context) error {
```
Comment (amyangfei) [inline cdc/owner.go:297]: This is a compatibility hack between v4.0.0 and v4.0.1, leave a comment about this, as it is not a general resource clean API

File: cdc/owner.go [hunk 3]
```diff
@@ -294,6 +294,47 @@ func (o *Owner) newChangeFeed(
...
+			err := o.cfRWriter.RemoveAllTaskStatus(ctx, changefeedID)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			cleaned = true
+		case nil:
+		default:
+			return errors.Trace(err)
+		}
+		_, err = o.cfRWriter.GetAllTaskPositions(ctx, changefeedID)
```
Comment (amyangfei) [inline cdc/owner.go:316]: Does `/tidb/cdc/task/position/<>/<>` have compatibility changes?
If not, the task position will be retained

**Comment by leoppro** [comment]
@amyangfei PTAL again

**Pull request merged**